packages/iov-lisk/src/liskclient.ts

```diff
@@ -0,0 +1,200 @@
+import axios from "axios";
+import { ReadonlyDate } from "readonly-date";
+import { Stream } from "xstream";
+
+import {
+  Address,
+  BcpAccount,
+  BcpAccountQuery,
+  BcpAddressQuery,
+  BcpNonce,
+  BcpQueryEnvelope,
+  BcpTicker,
+  BcpTransactionResponse,
+  ConfirmedTransaction,
+  IovReader,
+  Nonce,
+  TokenTicker,
+} from "@iov/bcp-types";
+import { Encoding } from "@iov/encoding";
+import { Algorithm, ChainId, PostableBytes, PublicKeyBytes, Tag, TxId, TxQuery } from "@iov/tendermint-types";
+
+import { Parse } from "./parse";
+
+function isAddressQuery(query: BcpAccountQuery): query is BcpAddressQuery {
+  return (query as BcpAddressQuery).address !== undefined;
+}
+
+/**
+ * Encodes the current date and time as a nonce
+ */
+export function generateNonce(): Nonce {
+  const now = new ReadonlyDate(ReadonlyDate.now());
+  return Parse.timeToNonce(now);
+}
+
+export class LiskClient implements IovReader {
+  private readonly baseUrl: string;
+
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl;
+  }
+
+  public disconnect(): void {
+    // no-op
+  }
+
+  public async chainId(): Promise<ChainId> {
+    const url = this.baseUrl + "/api/node/constants";
+    const result = await axios.get(url);
+    const responseBody = result.data;
+    return responseBody.data.nethash;
+  }
+
+  public async height(): Promise<number> {
+    const url = this.baseUrl + "/api/node/status";
+    const result = await axios.get(url);
+    const responseBody = result.data;
+    return responseBody.data.height;
+  }
+
+  public async postTx(bytes: PostableBytes): Promise<BcpTransactionResponse> {
+    const transactionId = JSON.parse(Encoding.fromUtf8(bytes)).id as string;
+    if (!transactionId.match(/^[0-9]+$/)) {
+      throw new Error("Invalid transaction ID");
+    }
+
+    await axios.post(this.baseUrl + "/api/transactions", bytes, {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    // Sleep some seconds to ensure transaction will be found.
+    // Sleep duration determined by trial and error. 15 seconds was not enough.
+    await new Promise(resolve => setTimeout(resolve, 20 * 1000));
```

Comment:
wait, so post always succeeds? just that it is in mempool, not that is it commited?
and you have to check the trasnactionid in the search to see if it made it...

hmmm... this code is fine for now, I can see a refinement coming up

Reply:
It succeeds except for
- malformed requests
- invalid transaction signatures
- insufficient funds

No idea what other checks are done. But yeah, we cannot know if the transaction makes it into the blockchain at this point. At a blocktime of 10 seconds and 25 transactions per block, it can take minutes until the transaction is in the chain if the transaction pool of full.

Reply:
there are many checks around vote list, multisig and a few other cases.